src/components/project/project.service.ts

```diff
@@ -380,6 +381,47 @@ export class ProjectService {
     return event.updated;
   }
 
+  async updateStep(
+    input: ProjectTransitionInput,
+    session: Session
+  ): Promise<UnsecuredDto<Project>> {
+    const currentProject = await this.readOneUnsecured(
+      input.id,
+      session,
+      input.changeset
+    );
+
+    const changes = this.repo.getActualChanges(currentProject, input);
+
+    if (!changes.step) {
+      return currentProject;
+    }
+
+    // verify new project step
+    await this.projectRules.verifyStepChange(
+      input.id,
+      session,
+      changes.step,
+      input.changeset
+    );
+
+    // update project step prop
+    const result = await this.repo.updateProperties(
+      currentProject,
+      { step: changes.step },
+      input.changeset
+    );
+
+    const event = new ProjectUpdatedEvent(
```

Comment:
I am thinking it will be worth creating a new event only for Project Step update (`ex: ProjectStepUpdatedEvent`)

Reply:
Would that actually simplify any of the handlers? I think not because they still have to check previous/current step to determine if they should run.